Adding clear logging including debug info (GPU)

juhlee-microsoft opened the pull request:
1. No functional change
2. Added more debug msg

Test results of CUDA driver
```
[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 11/12/2019 23:10:24
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:22
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 GPU                  NVIDIA-CUDA-DRIVER-VALIDATION                                                     PASS                19.83 
[Canonical UbuntuServer 18.04-LTS latest] 	Using nVidia driver : CUDA 
[Canonical UbuntuServer 18.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 4, count inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 4, found inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lshw: Expected Display adapters: 4, total adapters found in VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	nvidia-smi: Expected GPU count: 4, found inside the VM: 4 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\GZ90\TestResults\2019-12-11-23-10-20-9773
```
Test result of GRID driver
```
[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 11/12/2019 23:10:02
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1023-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:26
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 GPU                  NVIDIA-GRID-DRIVER-VALIDATION                                                     PASS                23.86 
[Canonical UbuntuServer 18.04-LTS latest] 	Using nVidia driver : GRID 
[Canonical UbuntuServer 18.04-LTS latest] 	lsvmbus: Expected "PCI Express pass-through" count: 1, count inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lspci: Expected "3D controller: NVIDIA Corporation" count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	lshw: Expected Display adapters: 1, total adapters found in VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 	nvidia-smi: Expected GPU count: 1, found inside the VM: 1 : PASS 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\NH54\TestResults\2019-12-11-23-09-58-9042
```

chvalean commented on Testscripts/Linux/dpdkSetup.sh:
```diff
@@ -118,12 +118,18 @@ function install_dpdk () {
-	ssh "${1}" "cd $HOMEDIR/$dpdkSrcDir/$DPDK_BUILD && make -j8 && make install"
+	proc_nr=$(cat /proc/cpuinfo | grep vendor | wc -l)
+	ssh "${1}" "cd $HOMEDIR/$dpdkSrcDir/$DPDK_BUILD && make -j${proc_nr} && make install"
 	check_exit_status "dpdk build on ${1}"
```
suggest to do a check_exit_status after make and another one after make install?
Then it can be clear to report if the compilation or the install failed.

ader1990 replied:
make install is just a shim call that copies some files to another folder, I don t think it is relevant to check each calls

ader1990 commented on Testscripts/Linux/dpdkSetup.sh:
```diff
@@ -118,12 +118,18 @@ function install_dpdk () {
-	ssh "${1}" "sed -i 's/^CONFIG_RTE_LIBRTE_MLX4_PMD=n/CONFIG_RTE_LIBRTE_MLX4_PMD=y/g' $HOMEDIR/$dpdkSrcDir/config/common_base"
-	check_exit_status "${1} CONFIG_RTE_LIBRTE_MLX4_PMD=y"
+	LogMsg "CONFIG_RTE_LIBRTE_MLX._PMD enable ${1}"
+	ssh "${1}" "sed -ri 's,(MLX._PMD=)n,\1y,' $HOMEDIR/$dpdkSrcDir/config/common_base"
```
@justin-parus @juhlee-microsoft these config changes are okay with you?
before the change, only mlx4_pmd was enabled, but I have seen in the azure dpdk docs that mlx5 and netvsc are also required.
Do you have any insight on this?

Thank you.

justin-parus replied:
I have started using dpdkSetupAndRunTest.sh in combination with dpdkUtils.sh as it provides a more complete set of tools to test dpdk. dpdkUtils.sh provides `Install_Dpdk()` which installs dpdk with support for both mlx 4,5 and follows the method from the official [doc](https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk).
```
ssh ${install_ip} "cd ${LIS_HOME}/${DPDK_DIR} && make config T=x86_64-native-linuxapp-gcc"
ssh ${install_ip} "sed -ri 's,(MLX._PMD=)n,\1y,' ${LIS_HOME}/${DPDK_DIR}/build/.config"
```

The files dpdkSetupAndRunTest.sh + dpdkUtils.sh + DPDK_TESTCASE_DRIVER.ps1 acts as a "micro"-framework within the LISAv2 in that it provides some boilerplate code for dpdk tests. dpdk-testPMD.sh, dpdk-testFWD.sh, and dpdk-testFailsafe.sh all use this setup. 

dpdkSetup.sh and dpdkTestPmd.sh were great starting points and much of the code in the newer dpdk files was derived from them. I think it would be okay to deprecate these older files. Check with @juhlee-microsoft to be sure they aren't being used elsewhere or aren't piping data to a PowerBI graph. Last I heard this was already moved to use dpdk-TestPMD.sh instead

justin-parus replied:
Further dpdkSetupAndRunTest.sh + dpdkUtils.sh + DPDK_TESTCASE_DRIVER.ps1 are in a functional state right now, but still could use much improvement and are open to suggestion. For example, what @chvalean mention above about checking the status of the build is something that has been needed for awhile now. As I have seen builds mysteriously fail (my guess is something with `make -j`) before and waste pipeline time.

ader1990 replied:
@justin-parus dpdkSetup.sh is still used by the test cases mentioned by @juhlee-microsoft 

To give more context, these are the changes required for building OVS/VPP/NFF-GO and ANS with DPDK support.
This is work in progress in order to make a dpdk setup which works for the new test cases planned for DPDK.

ader1990 replied:
@justin-parus My main concern is that these changes might affect the other test cases, though it should not be the case, which is why I wanted your insight on this matter.

juhlee-microsoft commented:
@ader1990 I know this is low priority now. Can you complete this PR or pull back? It's been 28 days.

ader1990 commented:
> @ader1990 I know this is low priority now. Can you complete this PR or pull back? It's been 28 days.

closing this for the moment, will sync with Stephen H. and make sure this is not required.

juhlee-microsoft committed:
Improved the logging in GPU tensorflow tests

juhlee-microsoft committed:
Followed up the feedback

1. Removed the rest of curly brace in bash, if unnecessary.
2. Fixed the wording
3. Removed the double $ in PS

juhlee-microsoft committed:
Update gpu-driver-install.sh

juhlee-microsoft committed:
Added more logging in utils.sh
